src/codewhisperer/activation.ts

```diff
@@ -202,6 +206,39 @@ export async function activate(context: ExtContext): Promise<void> {
         auth.showReauthenticatePrompt()
     }
 
+    function tryShowGettingStartedPage() {
+        //This logic is to show the Getting Started Page only once for the first time SignIn Users
+        let start = ''
+
+        /**
+         * hasShownWelcomeMsgBefore defines whether user is existing user or new user.
+         */
+        const hasShownWelcomeMsgBefore = get(
+            /** welcomeMessageKeyPrevious is a key for the previous welcome page of version v1.91.0,*/
+            CodeWhispererConstants.welcomeMessageKeyPrevious,
+            context.extensionContext.globalState
```

Comment:
`get` is a very common name so I would namespace it by importing the module name instead of the function itself.

Different question: instead of storing this in globalState, did CW team consider using the [PromptSettings](https://github.com/aws/aws-toolkit-vscode/blob/de8244c7172da804fdfc796914a7372b0c7c1d7a/src/shared/settings.ts#L530) feature? 

1. just add a new prompt kind [here](https://github.com/aws/aws-toolkit-vscode/blob/de8244c7172da804fdfc796914a7372b0c7c1d7a/package.json#L172)
2. then check if the prompt is enabled [like this](https://github.com/aws/aws-toolkit-vscode/blob/de8244c7172da804fdfc796914a7372b0c7c1d7a/src/codecatalyst/activation.ts#L88)

Reply:
Yes, `get` was an **existing function** in the code so I reused it.
I will change the name from "get" to "getState"

Reply:
Its name doesn't necessarily need to be changed. Just the import. Like this:

    import * as foo from '...'

then it can be used like this:

    foo.get()

that is easier to follow than un-namespaced. `get`. 

Renaming it to `getFoo()` is a different way of "namespacing".

Reply:
however, the main suggestion above is regarding "PromptSettings", not get().

Reply:
I need to look about "PromptSettings" neither I used nor explored on that.

Reply:
`oldWelcomeMessageKey` is a legacy thing. My comment is a general comment about the functionality, not about the "old message" specifically. It's true that PromptSettings can't be used for the "old" message. But it can be used for the "new message" functionality.

Reply:
From what I understood, PromptSettings currently are used per user signIn session. If we sign out and sign-In, this will restore the values which is not expected in this functionality.
**Expected:-**
If a user once signIn, Welcome Page is shown and thats it even if the user sign out and sign-in this should not be shown again.
**By using PromptSettings:-**
If a user sign-in, Welcome Page is displayed and if user sign out and sign-in, again the Welcome Page is shown because for each sign out and sign in the values reset to default values.

Reply:
> From what I understood, PromptSettings currently are used per user signIn session. If we sign out and sign-In, this will restore the values which is not expected in this functionality.

PromptSettings are stored in the user's vscode settings (settings.json). Not affected by sign out.

Reply:
To summarize:
Use the global state for the old(existing) state, since it's the place where the state is already stored by existing toolkit versions.
Use the PromptSettings for the new state

Reply:
updated accordingly.